console/src/main/scala/io/joern/console/cpgcreation/ImportCode.scala

```diff
@@ -171,22 +176,28 @@ class ImportCode[T <: Project](console: io.joern.console.Console[T])(implicit
       extends SourceBasedFrontend(name, Languages.NEWC, "Eclipse CDT Based Frontend for C/C++", extension)
 
   private def withCodeInTmpFile(str: String, filename: String)(f: File => Cpg): Try[Cpg] = {
-    val dir = File.newTemporaryDirectory("console")
+    val dir = FileUtil.newTemporaryDirectory("console")
+
     val result = Try {
-      (dir / filename).write(str)
+      Files.write((dir / filename).toPath, str.getBytes(Charset.defaultCharset()))
```

Comment:
Also check out. Files.writeString
You may need to add StandardOpenOptions to truncate/create etc.

Reply:
Yes, please use `Files.writeString` it is just shorter. It is a Java 11 API  but that is fine. We anyway already depend on other Java 11 APIs.